src/pineko/evolve.py

```diff
@@ -103,9 +105,11 @@ def write_operator_card_from_file(
     if not pathlib.Path(pineappl_path).exists():
         raise FileNotFoundError(pineappl_path)
     pineappl_grid = pineappl.grid.Grid.read(pineappl_path)
+    pineappl_grid.optimize()
```

Comment:
Why is there an optimize here? I think this might be the cause of #228

Reply:
Yes, the problem was that another optimize was already present in pineko evolve (https://github.com/NNPDF/pineko/blob/0c01d8d480a16821bf1e24e3884420bb9d72f146/src/pineko/theory.py#L435-L436) , so I added one before the `opcard` generation because otherwise there were some mismatches. 
But I'm not sure why the grid in #228 can't be optimised.

Reply:
I'm not entirely sure how I feel about that. I would remove both optimize because ptherwise you cannot use directly pineappl evolve since the original grid and the ekos don't match.

Reply:
On a second thought (again), I agree with @felixhekhorn. In my comment above, I was still thinking of scenarios in which optimise could "modify" the grids (scenarios such as the initial states are different but optimised in the same way). But from v1 that will never happen anymore. So one **should** always optimise by default.

Reply:
> That said, practically I would either remove them or wrap them with try-except so we can move forward with the current version of the codes.

This is the short-term solution, I agree. However, mid-term we definitely want to fix the PineAPPL bug instead ...

> But wouldn't that mean that ekos created with pineko cannot be used with `pineappl evolve`? (as you would need to optimize to make sure the grids and channels match)

This would only happen in a scenario, which I consider unlikely: one has a grid, calls optimize without saving the grid, then extract the necessary ekos, and then use the original (unoptimized) grid with the ("optimized") EKO. I would not support such a thing - or if you want to, force directly an optimize inside `pineappl evolve`.

Reply:
> edit: We can enforce an o `optimize`

I was about to suggest the edit, i.e. I'm much in favour of it :innocent: 

Also as eko developer I would say, that computing an eko is expensive and worth to optimize :see_no_evil:

Reply:
> I don't think this would be a problem though! The EKO grids are Q 2 slices of a 4D array of ( f out , x out , f in , x in ) and PineAPPL doesn't require f in to match the grid. So, as long as x in matches, everything should work fine.

`optimize` can remove $Q^2$ which would be a problem for the CLI (but is the feature we=eko wants)